### PR TITLE
Remove metac/metap/metacp from the public API

### PR DESCRIPTION
**Pull request by xeno-by**
Having these guys in a non-internal package was an oversight.

### CODE REVIEW

**Comment by olafurpg** [comment]
I think both `main` and `process` should be part of the public API.

**Comment by xeno-by** [comment]
@olafurpg I agree that we need to have the entry points in the public API. 

But I propose that we hide everything else, including `process`. Unless programmatic in-process invocation of cli tools gets on our critical path, I don't think we should expand the public API to accommodate it.

**Comment by olafurpg** [comment]
I think process is nice to have in the public API, main is unusable as a library because it exits the jvm.

**Comment by xeno-by** [comment]
Sure. But do we want to expose unspecified stdout as the public API? The
only thing that can be reliably done with it is printing, but that works
just as well with forking.

**Comment by olafurpg** [comment]
I'd like to be able to use `coursier bootstrap metac --main scala.meta.cli.Metac` without it breaking. The `process` method is equally useful when invoking the API programmatically for example via reflection when you don't want to `sys.exit`, maybe some build tool might like that. We promise no compatibility in the output, just promise that these entrypoints will remain stable.

File: semanticdb/metacp/src/main/scala/scala/meta/cli/Metacp.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+object Metacp {
+  def main(args: Array[String]): Unit = {
+    Settings.parse(args.toList) match {
+      case Some(settings) => sys.exit(process(settings))
+      case None => sys.exit(1)
+    }
+  }
+
+  def process(settings: Settings): Int = {
```
[the remark this answers is not in the document]
Comment (xeno-by) [reply semanticdb/metacp/src/main/scala/scala/meta/cli/Metacp.scala]: Done.

File: tests/jvm/src/test/scala/scala/meta/tests/cli/CliSuite.scala [hunk 2]
```diff
@@ -5,8 +5,8 @@ import java.nio.charset.StandardCharsets._
 import java.nio.file._
 import scala.util.Properties.versionNumberString
 import org.scalatest.FunSuite
-import scala.meta.cli.metac.{Main => Metac}
-import scala.meta.cli.metap.{Main => Metap}
+import scala.meta.internal.metac.{Main => Metac}
```
Comment (olafurpg) [inline tests/jvm/src/test/scala/scala/meta/tests/cli/CliSuite.scala]: Are these renames necessary anymore?

**Pull request merged**